packages/vuelidate/src/core.js

```diff
@@ -307,7 +307,7 @@ function collectNestedValidationResults (validations, state, key, path, resultsC
   return nestedValidationKeys.reduce((results, nestedKey) => {
     // if we have a key, use the nested state
     // else use top level state
-    const nestedState = key ? state[key] : state
+    const nestedState = key ? computed(() => unwrap(unwrap(state)[key])) : state
```

Comment:
Why are we wrapping here in a computed property? Because you would pull the property out of the ref object, making it none reactive?

I wonder if we are wrapping it in a computed any way, do we need to unwrap it?

Reply:
Exactly, the computed is needed to preserve reactivity.

`computed` can not track dependencies if they are not unwrapped. There are 2 situations:
  1 - state is a ref (this is a deeply nested property and the parent is `computed`)
  2 - state[key] is a ref (from `useVuelidate(rules, { key })`)

We need to unwrap the refs for every possible case.